app/assets/javascripts/google_analytics.js

```diff
@@ -1,7 +1,9 @@
-window.dataLayer = window.dataLayer || [];
-function gtag() {
-  dataLayer.push(arguments);
-}
-gtag("js", new Date());
+if (window.TeacherPayments.cookies.checkNonEssentialCookiesAccepted()) {
+  window.dataLayer = window.dataLayer || [];
+  function gtag() {
+    dataLayer.push(arguments);
+  }
+  gtag("js", new Date());
 
-gtag("config", document.currentScript.getAttribute("data-ga-id"), { anonymize_ip: true });
+  gtag("config", document.currentScript.getAttribute("data-ga-id"), { anonymize_ip: true });
+}
```

Comment:
I think if you ran this on DOM loaded, you'd be able to rely on the cookie functions without taking them out of the build system. Or put the cookies code before `= require_tree .`.